### ISSUE TITLE
Class without `__init__` and `__new__` is  `(*args: Any, **kwargs: Any) -> Class`

### ISSUE THREAD

**Issue by juk-r** [microsoft/pyright#6426]
The revealed type is `(*args: Any, **kwargs: Any) -> Foo`, but must be `() -> Foo`
```python
from typing import TypeVar, ParamSpec, Callable

P = ParamSpec("P")
T = TypeVar("T")

def cast_to_callable(cls: Callable[P, T]) -> Callable[P, T]:
    return cls

class Foo:
    pass

reveal_type(cast_to_callable(Foo))
```

**Comment by erictraut** [comment]
Yeah, I agree that it's OK to assume that the signature is empty (no parameters) if we know it's a specific class (it's not a `type[T]`) and it's not a protocol class. In those cases, I think it's best to retain the `(*args: Any, **kwargs: Any)` signature when converting the type to a callable.

**Comment by erictraut** [comment]
`(...)` and `(*args: Any, **kwargs: Any)` are equivalent in this context.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.336, which I just published. It will also be included in a future release of pylance.
